idx/memory/memory.go

```diff
@@ -1251,9 +1269,13 @@ func (m *MemoryIdx) Prune(oldest time.Time) ([]idx.Archive, error) {
 	pre := time.Now()
 
 	m.RLock()
+
+	// getting all checks once saves having to recompute the cutoff everytime we have a match
+	indexChecks := IndexRules.Checks(now)
```

Comment:
can we move this to before we acquire the RLock().  We want to keep lock times as small as possible

Reply:
oops